website_order_invoice_automation/models/sale_order_line.py

```diff
@@ -0,0 +1,40 @@
+# -*- coding: utf-8 -*-
+# Copyright 2018 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class SaleOrderLine(models.Model):
+    _inherit = 'sale.order.line'
+
+    auto_invoice = fields.Boolean(
+        related='order_id.team_id.auto_invoice',
+        store=True,
+    )
+
+    @api.depends('qty_invoiced', 'qty_delivered', 'product_uom_qty',
+                 'order_id.state', 'order_id.team_id.invoice_policy')
+    def _get_to_invoice_qty(self):
+        for line in self:
+            invoice_policy = line.order_id.team_id.invoice_policy or \
+                             line.product_id.invoice_policy
```

Comment:
Consideration on services (delivery charges, etc.) seems to be missing.

Reply:
The checking have been added.